c/src/sbp.c

```diff
@@ -227,13 +242,45 @@ s8 sbp_remove_callback(sbp_state_t *s, sbp_msg_callbacks_node_t *node)
   return SBP_CALLBACK_ERROR;
 }
 
+/** Register a callback for ANY message type.
+ * Register a callback that is called when any
+ * valid message is received.
+ *
+ * \param msg_type Message type associated with callback
+ * \param fn       Pointer to message callback function
+ * \param context  Pointer to context for callback function
+ * \return `SBP_OK` (0) if successful, `SBP_NULL_ERROR` if fn is NULL
+ *
+ */
+s8 sbp_register_allmsg_callback(sbp_state_t *s, sbp_allmsg_callback_fn_t fn,
+                                void *context) {
+  /* Check our callback function pointer isn't NULL. */
+  if (0 == fn) {
+    return SBP_NULL_ERROR;
+  }
+  s->allmsg_cb = fn;
+  s->allmsg_context = context;
+  return SBP_OK;
+}
+
+/** Remove allmsg function pointer.
+ *
+ * \return `SBP_OK` (0) if successful
+ */
+s8 sbp_remove_allmsg_callback(sbp_state_t *s) {
+  s->allmsg_cb = 0;
+  return SBP_OK;
+}
+
 /** Clear all registered callbacks.
  * This is probably only useful for testing but who knows!
  */
 void sbp_clear_callbacks(sbp_state_t *s)
 {
   /* Reset the head of the callbacks list to NULL. */
   s->sbp_msg_callbacks_head = 0;
+  /* Reset the all_cb to 0. */
+  s->allmsg_cb = 0;
```

Comment:
I would argue this shouldn't be included, `sbp_remove_allmsg_callback` already covers clearing this

Reply:
I was debating that as well.  If I don't use the word "callback" for the "allmsg_fuction", then this would be very clear and I wouldn't need to change sbp_clear_callbacks.

Reply:
after renaming I agree and changed.